### PR TITLE
Make shutdown and stop_waiting noops when appropriate

### PR DESCRIPTION
**Pull request by johann-koebbe**

### CODE REVIEW

File: lib/bunny_burrow/base.rb [hunk 1]
```diff
@@ -24,9 +24,11 @@ def initialize
     end
 
     def shutdown
+      return if @shutdown
```
Comment (stlcardsfan) [inline lib/bunny_burrow/base.rb]: Does shutdown need to stop waiting too?

Comment (johann-koebbe) [inline lib/bunny_burrow/base.rb]: It shouldn't as shutdown calls stop_waiting. The smarts for whether to stop waiting or not is self-contained in stop_waiting. I didn't want shutdown to have to care about @waiting.

Comment (stlcardsfan) [inline lib/bunny_burrow/base.rb]: Ah, cool, I was missing that Server implemented its own shutdown with stop_waiting

**Comment by stlcardsfan** [comment]
Looks good

**Pull request merged**